docs/src/main/paradox/proto.md

```diff
@@ -0,0 +1,40 @@
+# Protobuf file format
+
+gRPC uses the Protobuf `.proto` file format to define your messages, services
+and some aspects of the code generation.
+
+For an in-depth description see the [official documentation](https://developers.google.com/protocol-buffers/docs/proto),
+but here are a few quick examples:
+
+## Messages
+
+Messages are defined in the same way protobuf definitions are used for serialization:
+
+@@snip [helloworld.proto]($root$/../plugin-tester-scala/src/main/protobuf/helloworld.proto) { #messages }
+
+## Services
+
+Those messages can be used to define services:
+
+@@snip [helloworld.proto]($root$/../plugin-tester-scala/src/main/protobuf/helloworld.proto) { #services }
+
+Both the request and the response consists of either a single message or a stream of messages.
+
+## Code generation options
+
+There are a number of options that can be set in the .proto definition that influence how the code is generated:
+
+@@snip [helloworld.proto]($root$/../plugin-tester-scala/src/main/protobuf/helloworld.proto) { #options }
+
+The 'package' in the `.proto` is used to resolve references from one `.proto` file to another.
+It can also be used for the package name in the generated code, but it is
+common to use the separate `java_package` option to override it.
```

Comment:
Is the photo package required or optional?

Reply:
Damn you, autocorrect!

Reply:
:camera: